tailormap-components/projects/core/src/lib/user-interface/attributelist/layer.service.ts

```diff
@@ -14,27 +16,22 @@ export class LayerService {
 
   // List of layers. The index is also the associated tab index.
   public layers: Layer[] = [];
+  private layersSubject = new BehaviorSubject<Layer[]>([]);
+  public layers$ = this.layersSubject.asObservable();
 
-  constructor(public tailorMapService: TailorMapService,
-              private ngZone: NgZone) {
+  constructor(private tailorMapService: TailorMapService) {
     // Install the layerVisibilityChanged handler.
     this.tailorMapService.layerVisibilityChanged$.subscribe(value => {
-      this.ngZone.run(() => {
-        this.loadLayers();
-      });
+      this.loadLayers();
     });
+    this.loadLayers();
   }
 
   public getAppId(): number {
     const vc = this.tailorMapService.getViewerController();
     return vc.app.id;
   }
 
-  public getLayers$(): Observable<Layer[]> {
-    this.loadLayers();
-    return of(this.layers);
```

Comment:
this returns an observable that is completed immediately. Rewrote that to a BehaviourSubject & Observable so subscriptions are actually updated once layers are updated.